tests/test_slashing.py

```diff
@@ -17,10 +17,79 @@ def test_slash_no_dbl_prepare(casper, funded_privkey, deposit_amount, new_epoch,
         casper.get_recommended_source_epoch(),
         funded_privkey
     )
-
+    old_deposit = casper.get_deposit_size(validator_index)
     casper.slash(vote_1, vote_2)
+    assert casper.get_deposit_size(validator_index) == old_deposit * 0.835
+
+def test_full_slash_no_dbl_prepare(casper, funded_privkey, deposit_amount, new_epoch,
+                              induct_validator, mk_vote, fake_hash, assert_tx_failed):
+    validator_index = induct_validator(funded_privkey, deposit_amount)
+    assert casper.get_total_curdyn_deposits() == deposit_amount
+
+    vote_1 = mk_vote(
+        validator_index,
+        casper.get_recommended_target_hash(),
+        casper.get_current_epoch(),
+        casper.get_recommended_source_epoch(),
+        funded_privkey
+    )
+    vote_2 = mk_vote(
+        validator_index,
+        fake_hash,
+        casper.get_current_epoch(),
+        casper.get_recommended_source_epoch(),
+        funded_privkey
+    )
+
+    initial_amount_deposited = casper.get_total_curdyn_deposits()
+    initial_amount_destroyed = casper.get_total_destroyed()
+
+    while (casper.get_total_destroyed()) < (int (initial_amount_deposited / 3)):
```

Comment:
Validator should only be able to be slashed once per epoch. Contract should remember if validator has been slashed and prevent subsequent slashing in the context of one epoch.

Should have a test that makes sure this _can't_ happen.

A test for full slash would need multiple validators.